File: leetcode/go/problem300.go
	func lengthOfLIS(nums []int) int {
		dp := make([]int, len(nums))
		dp[0] = 1
		mx := 1
		for i := 1; i < len(dp); i++ {
			dp[i] = 1
			for j := i - 1; j >= 0; j-- {
				if nums[j] < nums[i] {
					dp[i] = max(dp[j]+1, dp[i])
				}
			}
			mx = max(mx, dp[i])
		}
		return mx
	}

File: leetcode/python/problem300.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        m = {}

        def f(i):
            if i in m:
                return m[i]
            if i >= len(nums):
                return 0
            mx = 0
            for j in range(i, len(nums)):
                if nums[i] < nums[j]:
                    mx = max(mx, f(j))
            m[i] = mx + 1
            return mx + 1

        mx = 0
        for i in range(len(nums)):
            mx = max(mx, f(i))
        return mx
